### PR TITLE
Fix #505: Add info about Scala.js 0.6.x' End of Life.

### LINKED ISSUE #505: Please add information on 0.6.x End of Life
It would be great if there could be some information on the long life of 0.6.x and that it will no longer be maintained and whether any fixes will be accepted etc.

It might be good to add if support could be extended if someone wanted to fund bug fixes etc. via Scala Center funding if this is an option.

I think maybe the Introduction page may be good - it also has the Tutorial (0.6.x) on the right hand side that if you click says to use the 1.x version. The other option may be FAQ but that is at the bottom and not too visible.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: doc/project/dependencies.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ Use [scalajs-bundler](https://scalacenter.github.io/scalajs-bundler/) instead.
 Instead, it is provided by [sbt-jsdependencies](https://github.com/scala-js/jsdependencies).
 
 The rest of this section applies to Scala.js 0.6.x only.
+[Scala.js 0.6.x has reached End of Life.]({{ BASE_PATH }}/doc/internals/scalajs-0.6.x-eol.html)
```
Comment (gzm0) [inline doc/project/dependencies.md:33]: Instead of this, I propose we completely remove the "Deprecated: Depending on JavaScript libraries" section (and maybe add a basic intro to `scalajs-bundler`)?

I now realize that this is probably better done in another PR.

Comment (sjrd) [reply doc/project/dependencies.md:33]: Yes, more generally, I'd like to only show docs for 1.x by default, and have a checkbox to show information related to 0.6.x. A lightweight variant of versioned docs, in some sense.
